src/middlewared/middlewared/plugins/disk_/smart_attributes.py

```diff
@@ -38,7 +38,7 @@ async def smart_attributes(self, name):
         """
         Returns S.M.A.R.T. attributes values for specified disk name.
         """
-        output = json.loads(await self.middleware.call('disk.smartctl', name, ['-A', '-j']))
+        output = json.loads(await self.middleware.call('disk.smartctl', name, ['-A', '--json=c']))
 
         if 'ata_smart_attributes' in output:
```

Comment:
This method is only returning ATA based smart attributes but we allow returning smart attributes for SAS and NVMe. While you're here, you need to go ahead and add these drives types to this method.

Reply:
It doesn't look like SAS or NVMe drives report smart attribute information from smartctl sadly : (